x-pack/plugins/reporting/server/routes/diagnostic/browser.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ReportingCore } from '../..';
+import { API_DIAGNOSE_URL } from '../../../common/constants';
+import { browserTest } from '../../browsers/chromium/driver_factory/test';
+import { LevelLogger as Logger } from '../../lib';
+import { DiagnosticResponse } from '../../types';
+import { authorizedUserPreRoutingFactory } from '../lib/authorized_user_pre_routing';
+
+export const registerDiagnoseBrowser = (reporting: ReportingCore, logger: Logger) => {
+  const { router } = reporting.getPluginSetupDeps();
+  const userHandler = authorizedUserPreRoutingFactory(reporting);
+
+  router.post(
+    {
+      path: `${API_DIAGNOSE_URL}/browser`,
+      validate: {},
+    },
+    userHandler(async (user, context, req, res) => {
+      const body: DiagnosticResponse = await browserTest(reporting, logger);
```

Comment:
This is maybe a choice of style, but it might be a good idea to have `browserTest` reject the promise when there is an error. Then this could be wrapped in try/catch, and the `logger.error` messages in the browserTest module could be moved to the catch. 

As it is, `browserTest` resolves the promise even when there is an error. Like I said, it's maybe a choice of style, but when there is an error, it could `reject` with the Error object and you can decide what to do about errors in this handler.

Reply:
Ah, good point, I think it should propagate the error and have consumers handle, will adjust.